crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -254,6 +254,30 @@ pub fn impl_param_set(_input: TokenStream) -> TokenStream {
                 fn apply(&mut self, world: &mut World) {
                     self.0.apply(world)
                 }
+
+                fn world_access_level() -> WorldAccessLevel {
+                    let mut exclusive = false;
+                    let mut shared = false;
+                    #(
+                        match #param_fetch::world_access_level() {
+                            WorldAccessLevel::Exclusive => {
+                                exclusive = true;
+                            }
+                            WorldAccessLevel::Shared => {
+                                shared = true;
```

Comment:
Doesn't this need to panic if there's already exclusive access?

Reply:
Same story as the above. 

In `ParamSet`, the different params are allowed to conflict since you can only access one at a time, so the `ParamSet` just inherits the highest access level among them.

All the params have to be valid tho. If any of the params are invalid, `ParamSet` will panic since it checks the `world_access_level()` of each one.

Reply:
Ah, understandable, sorry

Reply:
Ah, understandable, sorry